psydac/fem/projectors.py

```diff
@@ -158,3 +157,258 @@ def get_dual_dofs(Vh, f, domain_h, backend_language="python", return_format='ste
         return tilde_f.toarray()
     else:
         return tilde_f
+
+#===============================================================================
+class DirichletBoundaryProjector(LinearOperator):
+
+    def __init__(self, fem_space, bcs=None, space_kind=None):
+
+        assert isinstance(fem_space, FemSpace)
+
+        coeff_space    = fem_space.coeff_space
+        self._domain   = coeff_space
+        self._codomain = coeff_space
+
+        if bcs is not None:
+            self._bcs = bcs
```

Comment:
Here I would add a type check, and make sure to store `_bcs` as a tuple:
```suggestion
        if bcs is not None:
            assert all(isinstance(bc, Boundary) for bc in bcs)
            self._bcs = tuple(bcs)
```
`Boundary` can be imported from `sympde.topology.basic`.